test/e2e/run_test.go

```diff
@@ -2456,4 +2457,102 @@ WORKDIR /madethis`, BB)
 	})
+
+	It("podman run log-opt overrides containers.conf path", func() {
+		expectedMessage := "CLI override test message"
+		confLogPath := filepath.Join(podmanTest.TempDir, "conf-logs")
+
+		conffile := filepath.Join(podmanTest.TempDir, "containers.conf")
+		configContent := fmt.Sprintf(`[containers]
+log_driver = "k8s-file"
+log_path = "%s"
+`, confLogPath)
+
+		err := os.WriteFile(conffile, []byte(configContent), 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = os.MkdirAll(confLogPath, 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		oldContainersConf := os.Getenv("CONTAINERS_CONF")
+		os.Setenv("CONTAINERS_CONF", conffile)
+		defer os.Setenv("CONTAINERS_CONF", oldContainersConf)
+
+		if IsRemote() {
+			podmanTest.RestartRemoteService()
+		}
+
+		cliLogPath := filepath.Join(podmanTest.TempDir, "cli-override.log")
+		session := podmanTest.Podman([]string{"run", "--rm", "--log-driver", "k8s-file", "--log-opt", fmt.Sprintf("path=%s", cliLogPath), ALPINE, "echo", expectedMessage})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		_, err = os.Stat(cliLogPath)
+		Expect(err).ToNot(HaveOccurred(), "Log should exist at CLI-specified path")
+
+		confLogDirs, err := os.ReadDir(confLogPath)
+		if err == nil { // The directory might not even be created, which is fine.
+			Expect(confLogDirs).To(BeEmpty(), "Config file log path should not be used when CLI overrides")
+		}
+
+		content, err := os.ReadFile(cliLogPath)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(string(content)).To(ContainSubstring(expectedMessage))
+		Expect(string(content)).To(MatchRegexp(`\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+.*stdout F ` + regexp.QuoteMeta(expectedMessage)))
+
+		_ = os.Remove(cliLogPath)
+	})
+
+	It("podman run uses containers.conf log_path", func() {
+		expectedMessage := "Config file path test message"
+		confLogPath := filepath.Join(podmanTest.TempDir, "conf-logs")
+
+		conffile := filepath.Join(podmanTest.TempDir, "containers.conf")
+		configContent := fmt.Sprintf(`[containers]
+log_driver = "k8s-file"
+log_path = "%s"
+`, confLogPath)
+
+		err := os.WriteFile(conffile, []byte(configContent), 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = os.MkdirAll(confLogPath, 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		oldContainersConf := os.Getenv("CONTAINERS_CONF")
+		os.Setenv("CONTAINERS_CONF", conffile)
+		defer os.Setenv("CONTAINERS_CONF", oldContainersConf)
+
+		if IsRemote() {
+			podmanTest.RestartRemoteService()
+		}
+
+		containerName := "test-conf-log-container"
+		session := podmanTest.Podman([]string{"run", "--name", containerName, ALPINE, "echo", expectedMessage})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+
+		time.Sleep(100 * time.Millisecond)
+
+		session = podmanTest.Podman([]string{"inspect", "--format", "{{.Id}}", containerName})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
```

Comment:
Oh FWIW please use `PodmanExitCleanly()` everywhere in new tests where possible which turns these three lines into one.